README.md

```diff
@@ -75,15 +76,16 @@ that you're trying to add to the collection isn't of the type `Integer`, a
 
 Fail fast.
 
-### Miscellaneous Collection with Integer Keys
+### Miscellaneous ArrayCollection with Integer Keys
 Similar to enforcing a type for the values, you can also fix the index (or key)
 types:
+
 ```php
 <?php declare(strict_types=1);
 
-use Managur\Collection\Collection;
+use Managur\Collection\ArrayCollection;
 
-final class IntegerKeyCollection extends Collection
+final class IntegerKeyCollection extends ArrayCollection
 {
     protected $keyType = 'integer';
```

Comment:
I'm wondering if these 'magic' key types could be extracted to a constant or something on the interface? I know 'integer' or 'string' isn't hard to remember, but much like magic numbers, magic strings don't sit right?

Reply:
That is an excellent idea! They can still be the string value underneath, but yeah, that's good. IDE auto-complete support included 😁

Reply:
Also, I should annotate with the enum options 👍